tests/test_lab_serve.py

```diff
@@ -109,6 +122,295 @@ def test_chat_manual(mock_popen, mock_determine, _, cli_runner: CliRunner):
+        [
+            f"--config={CFG_FILE_NAME}",
+            "model",
+            "serve",
+            "--model-path=foo",
+            "--",
+            "--tensor-parallel-size",
+            "4",
+        ],
+    )
+    assert args[15] == "--tensor-parallel-size"
+    assert args[16] == "4"
+    assert len(args) == 17
+
+
+@patch("time.sleep", side_effect=Exception("Intended Abort"))
+@patch("instructlab.model.backends.backends.determine_backend")
+@patch("subprocess.Popen")
+def test_gpus_with_ctx_tps_with_gpus_config(
+    mock_popen, mock_determine, _, cli_runner: CliRunner
+):
+    setup_gpus_config(gpus=8)
+    args = vllm_setup_test(
+        mock_popen,
+        mock_determine,
+        cli_runner,
+        [
+            f"--config={CFG_FILE_NAME}",
+            "model",
+            "serve",
+            "--model-path=foo",
+            "--gpus",
+            "2",
+            "--",
+            "--tensor-parallel-size",
+            "4",
+        ],
+    )
+    assert args[15] == "--tensor-parallel-size"
+    assert args[16] == "4"
+    assert len(args) == 17
+
+
+@patch("time.sleep", side_effect=Exception("Intended Abort"))
+@patch("instructlab.model.backends.backends.determine_backend")
+@patch("subprocess.Popen")
+def test_vllm_args_config(mock_popen, mock_determine, _, cli_runner: CliRunner):
+    setup_gpus_config(tps=8)
+    args = vllm_setup_test(
+        mock_popen,
+        mock_determine,
+        cli_runner,
+        [
+            f"--config={CFG_FILE_NAME}",
+            "model",
+            "serve",
+            "--model-path=foo",
+        ],
+    )
+    assert args[13] == "--tensor-parallel-size"
+    assert args[14] == "8"
+    assert len(args) == 15
+
+
+@patch("time.sleep", side_effect=Exception("Intended Abort"))
+@patch("instructlab.model.backends.backends.determine_backend")
+@patch("subprocess.Popen")
+def test_vllm_args_config_with_gpus_config(
+    mock_popen, mock_determine, _, cli_runner: CliRunner
+):
+    setup_gpus_config(gpus=4, tps=8)
+    args = vllm_setup_test(
+        mock_popen,
+        mock_determine,
+        cli_runner,
+        [
+            f"--config={CFG_FILE_NAME}",
+            "model",
+            "serve",
+            "--model-path=foo",
+        ],
+    )
+    assert args[15] == "--tensor-parallel-size"
+    assert args[16] == "4"
+    assert len(args) == 17
+
+
+@patch("time.sleep", side_effect=Exception("Intended Abort"))
+@patch("instructlab.model.backends.backends.determine_backend")
+@patch("subprocess.Popen")
+def test_vllm_args_config_with_gpus(
+    mock_popen, mock_determine, _, cli_runner: CliRunner
+):
+    setup_gpus_config(tps=8)
+    args = vllm_setup_test(
+        mock_popen,
+        mock_determine,
+        cli_runner,
+        [
+            f"--config={CFG_FILE_NAME}",
+            "model",
+            "serve",
+            "--model-path=foo",
+            "--gpus",
+            "4",
+        ],
+    )
+    assert args[15] == "--tensor-parallel-size"
```

Comment:
(no action needed) these asserts are a bit brittle, but I think we'll deal with them when we refactor the code that calculates cmd out into a function that we would be able to target properly.

Reply:
Agree on the brittleness.  Checking the length at least makes sure we are targeting the effective args.

Reply:
It's still not great but at least made it a little less brittle by using len(args) to calc the index.